client/components/app.jsx

```diff
@@ -6,31 +6,62 @@ import CarouselPic from './carousel.jsx';
 import DisplayPic from './display.jsx';
 import LeftArrow from './leftArrow.jsx';
 import RightArrow from './rightArrow.jsx';
+import HeartButton from './heartButton.jsx';
 import Modal from './modal.jsx';
 
 class App extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      itemName: 'BTS BT21 Official Pyjamas Set',
-      images: [
-        'https://btetsy.s3.us-east-2.amazonaws.com/2_1.jpg',
-        'https://btetsy.s3.us-east-2.amazonaws.com/2_2.jpg',
-        'https://btetsy.s3.us-east-2.amazonaws.com/2_3.jpg'
-      ],
+      images: [],
       currIndex: 0,
       translateVal: 0,
-      show: false
+      show: false,
+      like: false,
+      productId: null
     };
     this.nextPicture = this.nextPicture.bind(this);
     this.prevPicture = this.prevPicture.bind(this);
     this.selectedPic = this.selectedPic.bind(this);
     this.toggleModal = this.toggleModal.bind(this);
+    this.toggleHeart = this.toggleHeart.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
+
   }
+
   toggleModal (event) {
     this.setState({ show: !this.state.show})
   }
 
+  async toggleHeart (event) {
+    await this.setState({ like: !this.state.like});
```

Comment:
Is `this.setState` is `await`-able?

Reply:
Yes, it is await-able.

Reply:
@silkyh13 Ah, poorly worded question; everything in JavaScript is `await`able.

What benefit does `await`ing `this.setState` offer? I am not aware of a Promise API for `this.setState`.

Reply:
I tried it with both with and without await and it works with await. When await is not there, my put request does not update properly.